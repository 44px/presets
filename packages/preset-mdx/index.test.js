const plugin = require('./');

describe('plugin', () => {
  it('should merge to empty initial config', () => {
    const result = plugin.webpack();
    expect(result).toMatchSnapshot();
  });

  it('should merge to initial config with a few things', () => {
    const webpack = {
      entry: {},
      module: {
        rules: [
          {
            test: /\.css/,
            use: 'style-loader',
          },
        ],
      },
      optimization: {},
    };

    const result = plugin.webpack(webpack);
    expect(result).toMatchSnapshot();
  });

  it('should override "test" when rule is defined', () => {
    const webpack = {
      entry: {},
      module: {
        rules: [
          {
            test: /\.css/,
            use: 'style-loader',
          },
        ],
      },
      optimization: {},
    };

    const options = {
      rule: {
        test: /(.mdx)|(.mdy)?$/,
        exclude: ['foo.mdx'],
      },
    };

    const result = plugin.webpack(webpack, options);
    expect(result).toMatchSnapshot();
  });

  it('should add babelLoaderOptions when is defined', () => {
    const webpack = {
      entry: {},
      module: {
        rules: [
          {
            test: /\.css/,
            use: 'style-loader',
          },
        ],
      },
      optimization: {},
    };

    const options = {
      babelLoaderOptions: {
        babelrc: false,
      },
    };

    const result = plugin.webpack(webpack, options);
    expect(result).toMatchSnapshot();
  });

  it('should exclude babel loader when babelLoaderOptions = false', () => {
    const webpack = {
      entry: {},
      module: {
        rules: [
          {
            test: /\.css/,
            use: 'style-loader',
          },
        ],
      },
      optimization: {},
    };

    const options = {
      babelLoaderOptions: false,
    };

    const result = plugin.webpack(webpack, options);
    expect(result).toMatchSnapshot();
  });

  it('should combine mdx loader options when mdxLoaderOptions is defined', () => {
    const webpack = {
      entry: {},
      module: {
        rules: [
          {
            test: /\.css/,
            use: 'style-loader',
          },
        ],
      },
      optimization: {},
    };

    const options = {
      mdxLoaderOptions: {
        mdPlugins: ['a'],
        hastPlugins: ['b'],
        compilers: ['c'],
      },
    };

    const result = plugin.webpack(webpack, options);
    expect(result).toMatchSnapshot();
  });
});
