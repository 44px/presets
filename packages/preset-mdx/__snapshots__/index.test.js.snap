// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin should add babelLoaderOptions when is defined 1`] = `
Object {
  "entry": Object {},
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.css/,
        "use": "style-loader",
      },
      Object {
        "test": /\\.mdx\\?\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
            },
          },
          Object {
            "loader": "@mdx-js/loader",
            "options": Object {
              "compilers": Array [
                [Function],
              ],
              "hastPlugins": Array [
                [Function],
              ],
              "mdPlugins": Array [
                Array [
                  [Function],
                  undefined,
                ],
                Array [
                  [Function],
                  undefined,
                ],
              ],
            },
          },
        ],
      },
    ],
  },
  "optimization": Object {},
}
`;

exports[`plugin should combine mdx loader options when mdxLoaderOptions is defined 1`] = `
Object {
  "entry": Object {},
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.css/,
        "use": "style-loader",
      },
      Object {
        "test": /\\.mdx\\?\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
          },
          Object {
            "loader": "@mdx-js/loader",
            "options": Object {
              "compilers": Array [
                "c",
                [Function],
              ],
              "hastPlugins": Array [
                "b",
                [Function],
              ],
              "mdPlugins": Array [
                Array [
                  [Function],
                  undefined,
                ],
                Array [
                  [Function],
                  undefined,
                ],
                "a",
              ],
            },
          },
        ],
      },
    ],
  },
  "optimization": Object {},
}
`;

exports[`plugin should exclude babel loader when babelLoaderOptions = false 1`] = `
Object {
  "entry": Object {},
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.css/,
        "use": "style-loader",
      },
      Object {
        "test": /\\.mdx\\?\\$/,
        "use": Array [
          Object {
            "loader": "@mdx-js/loader",
            "options": Object {
              "compilers": Array [
                [Function],
              ],
              "hastPlugins": Array [
                [Function],
              ],
              "mdPlugins": Array [
                Array [
                  [Function],
                  undefined,
                ],
                Array [
                  [Function],
                  undefined,
                ],
              ],
            },
          },
        ],
      },
    ],
  },
  "optimization": Object {},
}
`;

exports[`plugin should merge to empty initial config 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\.mdx\\?\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
          },
          Object {
            "loader": "@mdx-js/loader",
            "options": Object {
              "compilers": Array [
                [Function],
              ],
              "hastPlugins": Array [
                [Function],
              ],
              "mdPlugins": Array [
                Array [
                  [Function],
                  undefined,
                ],
                Array [
                  [Function],
                  undefined,
                ],
              ],
            },
          },
        ],
      },
    ],
  },
}
`;

exports[`plugin should merge to initial config with a few things 1`] = `
Object {
  "entry": Object {},
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.css/,
        "use": "style-loader",
      },
      Object {
        "test": /\\.mdx\\?\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
          },
          Object {
            "loader": "@mdx-js/loader",
            "options": Object {
              "compilers": Array [
                [Function],
              ],
              "hastPlugins": Array [
                [Function],
              ],
              "mdPlugins": Array [
                Array [
                  [Function],
                  undefined,
                ],
                Array [
                  [Function],
                  undefined,
                ],
              ],
            },
          },
        ],
      },
    ],
  },
  "optimization": Object {},
}
`;

exports[`plugin should override "test" when rule is defined 1`] = `
Object {
  "entry": Object {},
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.css/,
        "use": "style-loader",
      },
      Object {
        "exclude": Array [
          "foo.mdx",
        ],
        "test": /\\(\\.mdx\\)\\|\\(\\.mdy\\)\\?\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
          },
          Object {
            "loader": "@mdx-js/loader",
            "options": Object {
              "compilers": Array [
                [Function],
              ],
              "hastPlugins": Array [
                [Function],
              ],
              "mdPlugins": Array [
                Array [
                  [Function],
                  undefined,
                ],
                Array [
                  [Function],
                  undefined,
                ],
              ],
            },
          },
        ],
      },
    ],
  },
  "optimization": Object {},
}
`;
